ql/math/array.hpp

```diff
@@ -145,11 +145,9 @@ namespace QuantLib {
 
     //! specialization of null template for this class
     template <>
-    class Null<Array> {
-      public:
-        Null() = default;
-        operator Array() const { return Array(); }
-    };
+    inline Array Null<Array>() {
+        return Array();
```

Comment:
Stylistically it would be better to use the braced initializer list in all of these specializations as per https://clang.llvm.org/extra/clang-tidy/checks/modernize-return-braced-init-list.html.

Reply:
You can work around this by turning

    explicit Array(Size size = 0);

into

    Array();
    explicit Array(Size size);

Reply:
Nice catch - we have just pushed this change.